From dc1fae07d2896d9eace61989b7c25d9434ee7f3f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20=C5=BBo=C5=82nowski?= <aidecoe@aidecoe.name>
Date: Fri, 25 Nov 2011 10:08:47 +0100
Subject: [PATCH] Build symbol-test with make instead of hardcoding in
 symbol-hiding.

If symbol-test is built in symbol-hiding with hardcoded g++ invokation, it's
not so easy to pass $(srcdir) which is required to find notmuch.h when srcdir
and builddir are separate directories.

Fixed warnings in symbol-test.cc, btw.
---
 test/.gitignore     |    1 +
 test/Makefile.local |    5 ++++-
 test/basic          |    2 +-
 test/symbol-hiding  |    3 +--
 test/symbol-test.cc |    9 ++++-----
 5 files changed, 11 insertions(+), 9 deletions(-)

diff --git a/test/.gitignore b/test/.gitignore
index 9e97052..7e30e8d 100644
--- a/test/.gitignore
+++ b/test/.gitignore
@@ -1,4 +1,5 @@
 test-results
 corpus.mail
 smtp-dummy
+symbol-test
 tmp.*
diff --git a/test/Makefile.local b/test/Makefile.local
index 8eb0433..a672fd3 100644
--- a/test/Makefile.local
+++ b/test/Makefile.local
@@ -11,8 +11,11 @@ smtp_dummy_modules = $(smtp_dummy_srcs:.c=.o)
 $(dir)/smtp-dummy: $(smtp_dummy_modules)
 	$(call quiet,CC) $^ -o $@
 
+$(dir)/symbol-test: $(dir)/symbol-test.o
+	$(call quiet,CC) $^ -o $@ -Llib -lnotmuch -lxapian
+
 .PHONY: test check
-test:	all $(dir)/smtp-dummy
+test:	all $(dir)/smtp-dummy $(dir)/symbol-test
 	@${dir}/notmuch-test $(OPTIONS)
 
 check: test
diff --git a/test/basic b/test/basic
index 38db2ba..5463bf8 100755
--- a/test/basic
+++ b/test/basic
@@ -56,7 +56,7 @@ tests_in_suite=$(for i in $TESTS; do echo $i; done | sort)
 available=$(ls -1 $TEST_DIRECTORY/ | \
     sed -r -e "/^(aggregate-results.sh|Makefile|Makefile.local|notmuch-test)/d" \
 	   -e "/^(README|test-lib.sh|test-lib.el|test-results|tmp.*|valgrind|corpus*)/d" \
-	   -e "/^(emacs.expected-output|smtp-dummy|smtp-dummy.c|test-verbose|symbol-test.cc)/d" \
+	   -e "/^(emacs.expected-output|smtp-dummy|smtp-dummy.c|test-verbose|symbol-test.cc|symbol-test|symbol-test.o)/d" \
 	   -e "/^(test.expected-output|.*~)/d" \
 	   -e "/^(gnupg-secret-key.asc)/d" \
 	   -e "/^(gnupg-secret-key.NOTE)/d" \
diff --git a/test/symbol-hiding b/test/symbol-hiding
index d0b31ae..f67b653 100755
--- a/test/symbol-hiding
+++ b/test/symbol-hiding
@@ -12,13 +12,12 @@ test_description='exception symbol hiding'
 . ./test-lib.sh
 
 run_test(){
-    result=$(LD_LIBRARY_PATH=$TEST_DIRECTORY/../lib ./symbol-test 2>&1)
+    result=$(LD_LIBRARY_PATH=$TEST_DIRECTORY/../lib $TEST_DIRECTORY/symbol-test 2>&1)
 }
 
 output="A Xapian exception occurred opening database: Couldn't stat 'fakedb/.notmuch/xapian'
 caught No chert database found at path \`./nonexistant'"
 
-g++ -o symbol-test -I$TEST_DIRECTORY/../lib $TEST_DIRECTORY/symbol-test.cc -L$TEST_DIRECTORY/../lib -lnotmuch -lxapian
 mkdir -p fakedb/.notmuch
 test_expect_success 'running test' run_test
 test_begin_subtest 'checking output'
diff --git a/test/symbol-test.cc b/test/symbol-test.cc
index 1de06ea..bfbe38f 100644
--- a/test/symbol-test.cc
+++ b/test/symbol-test.cc
@@ -1,14 +1,13 @@
 #include <stdio.h>
 #include <xapian.h>
 #include <notmuch.h>
-main (int argc, char **argv){
+int main (){
 
-    notmuch_database_t *notmuch
-      = notmuch_database_open ("fakedb",
-				     NOTMUCH_DATABASE_MODE_READ_ONLY);
+  //notmuch_database_t *notmuch =
+  notmuch_database_open ("fakedb", NOTMUCH_DATABASE_MODE_READ_ONLY);
 
   try{
-    (void)new Xapian::WritableDatabase ("./nonexistant",					Xapian::DB_OPEN);
+    (void)new Xapian::WritableDatabase ("./nonexistant", Xapian::DB_OPEN);
   } catch (const Xapian::Error &error) {
     printf("caught %s\n",error.get_msg().c_str());
     return 0;
-- 
1.7.8.rc3

